ibis/tests/all/test_window.py

```diff
@@ -142,7 +150,14 @@ def test_window(backend, analytic_alltypes, df, con, result_fn, expected_fn):
             'Backend {} does not support window operations'.format(backend)
         )
 
-    expr = analytic_alltypes.mutate(value=result_fn)
+    expr = analytic_alltypes.mutate(
+        value=result_fn(
+            analytic_alltypes,
+            win=ibis.window(
+                following=0, group_by=["string_col"], order_by=["id"]
+            ),
+        )
+    )
```

Comment:
maybe would be good to move up line 165 `result = expr.execute().set_index('id').sort_index()` to just after here,  so if there is any windows operation that is unsupported, it don't need to run the following code.

Reply:
Good idea, done.